index.js

```diff
@@ -179,21 +179,43 @@ program
 
         if (onePagePath) {
           const lazyReloadMiddleware = function (req, res, next) {
-            const isHtmlFile = req.url.endsWith('.html');
+            const requestExtension = path.posix.extname(req.url);
 
-            if (isHtmlFile) {
-              const isDynamicIncludeHtmlFile = req.url.endsWith('._include_.html');
+            const hasEndingSlash = req.url.endsWith('/');
+            const hasNoExtension = requestExtension === '';
+            const isHtmlFileRequest = requestExtension === '.html' || hasEndingSlash || hasNoExtension;
 
-              if (!isDynamicIncludeHtmlFile) {
-                const urlWithoutBaseUrl = req.url.replace(config.baseUrl, '');
-                const urlWithoutExtension = fsUtil.removeExtension(urlWithoutBaseUrl);
+            if (!isHtmlFileRequest || req.url.endsWith('._include_.html')) {
+              next();
+              return;
+            }
+
+            let normalizedUrl = req.url.replace(config.baseUrl, '');
```

Comment:
We don't use `normalizedUrl` until L208. Could we define `normalizedUrl` closer to it's usage?

Also, I generally prefer using `const` and naming intermediate variables explicitly. I find that this makes it easier to follow what each step is doing.

Reply:
Good catch 😅, thanks! Updated some of the comments as well